pkg/components/pluggable/errors.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluggable
+
+import (
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type ErrorConverter func(status.Status) error
+
+// Compose together two errors converters by applying the inner first and if the error was not converted, then it applies to the outer.
+func (outer ErrorConverter) Compose(inner ErrorConverter) ErrorConverter {
+	return func(s status.Status) error {
+		err := inner(s)
+		st, ok := status.FromError(err)
```

Comment:
So an error is only _valid_ if it has an status message, right? That is what I got from here and `TestComposeErrorsConverters`